src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionService.cs

```diff
@@ -21,6 +22,99 @@ public DefaultTagHelperCompletionService(TagHelperFactsService tagHelperFactsSer
             _tagHelperFactsService = tagHelperFactsService;
         }
 
+        public override AttributeCompletionResult GetAttributeCompletions(AttributeCompletionContext completionContext)
+        {
+            if (completionContext == null)
+            {
+                throw new ArgumentNullException(nameof(completionContext));
+            }
+
+            var attributeCompletions = completionContext.ExistingCompletions.ToDictionary(
+                completion => completion,
+                _ => new HashSet<BoundAttributeDescriptor>(),
+                StringComparer.OrdinalIgnoreCase);
+
+            var documentContext = completionContext.DocumentContext;
+            var descriptorsForTag = _tagHelperFactsService.GetTagHelpersGivenTag(documentContext, completionContext.CurrentTagName, completionContext.CurrentParentTagName);
+            if (descriptorsForTag.Count == 0)
+            {
+                // If the current tag has no possible descriptors then we can't have any additional attributes.
+                var defaultResult = AttributeCompletionResult.Create(attributeCompletions);
+                return defaultResult;
+            }
+
+            var prefix = documentContext.Prefix ?? string.Empty;
+            Debug.Assert(completionContext.CurrentTagName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase));
```

Comment:
An assert because if the tag name doesn't start with the prefix it means there's a bug with GetTagHelpersGivenTag above.